File: ticketProject5-DistributedC/TicketSystemCC/model/model/Validators/SellerValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model.Validators
{
    /**
     * Validates an instance of class Seller
     * Throws ValidateException if instance is invalid
     */
    public class SellerValidator : Validator<Seller>
    {
        virtual public void Validate(Seller entity)
        {
            if (entity == null)
                throw new ArgumentNullException("entity must be not null");
            else
            {
                if (entity.Id <= 0)
                    throw new ValidationException("Invalid ID!");
                if (entity.FirstName == null)
                    throw new ValidationException("Invalid first name!");
                if (entity.LastName == null)
                    throw new ValidationException("Invalid last name!");
                if (entity.Username == null)
                    throw new ValidationException("Invalid username!");
                if (entity.Password == null)
                    throw new ValidationException("Invalid password!");
            }
        }
    }
}

File: ticketProject6-RMI/TicketSystem/TicketModel/src/main/java/ticket/model/validators/SellerValidator.java
package ticket.model.validators;

import ticket.model.Seller;

import java.security.InvalidParameterException;

/**
 * Validates an instance of class Seller
 * Throws ValidateException if instance is invalid
 */
public class SellerValidator implements Validator<Seller> {

    /**
     * @throws InvalidParameterException if id  id is null or <=0 / first name/last name are null.
     */
    @Override
    public void validate(Seller entity) throws ValidationException {
        if (entity == null)
            throw new InvalidParameterException();
        else {
            if (entity.getId() <= 0)
                throw new ValidationException("Invalid ID!");
            if (entity.getFirstName() == null)
                throw new ValidationException("Invalid first name!");
            if (entity.getLastName() == null)
                throw new ValidationException("Invalid last name!");
            if (entity.getUsername() == null || entity.getUsername().equals(""))
                throw new ValidationException("Invalid username!");
            if (entity.getPassword() == null)
                throw new ValidationException("Invalid password!");
        }
    }
}
